### DISCUSSION TITLE
Editor: Introduce a Track Changes mode

### DISCUSSION THREAD

**Discussion by tosteiner** [knowledgefutures/pubpub#1567] [Ideas]
A frequent question that we at COPIM get asked when introducing new community members to PubPub's collaborative writing features is about a "Track Changes" mode in the editor, i.e. a before-after visualization of text/content that is getting changed by subsequent user input in a collaborative writing scenario (see LibreOffice/MS Word/GoogleDocs editor).

As I also think this would be really neat to have, I just thought I'd ask if something like that would be on the development horizon of PubPub? This would be a major improvement that would help to convince more people to switch from gDocs to PubPub 😃

**Comment by gabestein** [comment]
Thanks Tobias, and sorry for the delay getting back to you. Building this is definitely on our radar (and [roadmap](https://github.com/orgs/pubpub/projects/9#card-52162899)), but I don't know exactly when we'll be able to commit to it. Right now we're mostly focused on submissions and review.

**Reply by publicityreform** [reply]
+1 a diff like feature to track content changes between releases would be wonderful. thank you!

**Reply by johav** [reply]
Hey @gabestein any news on when this can be added? We (AfricArXiv) have the same request :)

**Reply by gabestein** [reply]
Hi @johav! We actually have a working reference implementation and are in the process of scoping a first version of it as we speak. More very soon!

**Reply by tosteiner** [reply]
oh, wow, nice! This'll be a game-changer for collab. writing on PubPub, thanks so much for working on this, Gabe and team! 👏🏽

**Comment by amdeis** [comment]
Very excited for this to possibly include the ability to keep a member removed from the actual published version draft—so that they can't edit the draft at all without the managing member knowing (maybe there will be another member status? _view, edit, manage, admin_ are the current ways to add members, but maybe another for specifically tracked changes edits?) Would be especially great to have a share link (currently "view" and "edit" share link copy buttons) for tracked changes only. Excited about this!

**Reply by gabestein** [reply]
Us too, and thanks for the idea. In the first release, unfortunately, we're not going to be able to separate out the permissions, so track changes will be part of edit permissions. But that's on our list for next steps for the feature.

**Comment by MaPeras** [comment]
Any news on this feature? Cant wait for it to happen.

**Reply by tefkah** [reply]
Seems like it is being worked on? Unless "suggested edits" is something very different https://github.com/pubpub/pubpub/compare/master...kalilsn/suggested-edits

**Comment by gabestein** [comment]
Yup, it's in active development! Hoping for an initial release in a few weeks.

**Comment by MaPeras** [comment]
Thank you for making this possible! I just tried to co-write a paper via track changes mode. It worked ok, but two features would make it even better.

The ability to comment on proposed text (accept/decline is sometimes not a straightforward thing among academics, as we like to discuss until forever?).

Different colors for different authors, otherwise it gets really messy.

Cheers,
Max
